src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -248,6 +242,12 @@ HashSet<ITypeSymbol> typeStack
 				return MutabilityInspectionResult.NotMutable();
 			}
 
+			var scope = type.GetImmutabilityScope();
+			if( !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute ) && scope != ImmutabilityScope.None ) {

```

Comment:
If you're looking at a concrete type and its definition has either `[ImmutableBaseClass]` or `[Immutable]` then we're good (i.e. `[Immutable]` => `[ImmutableBaseClass]`)